src/vde_backend/compute_power.py

```diff
@@ -136,25 +137,39 @@ def get_consumption_production_df(raw_df, sensors_config):
     return cons_prod_df
 
 
-def get_home_consumption_production_df(home_rawdata, home_id, sensors_df):
+def get_home_consumption_production_df(home_raw_data, home_config):
     """
     compute power data from raw data (coming from cassandra 'raw' table) :
     P_cons = P_tot - P_prod
     P_net = P_prod + P_cons
-    """
 
-    first_sid = list(home_rawdata.keys())[0]
-    cons_prod_df = home_rawdata[first_sid][["sensor_id", "day", "ts"]].copy()
-    cons_prod_df = cons_prod_df.rename(columns={"sensor_id": "home_id"})
-    cons_prod_df["home_id"] = home_id  # replace 1st sensor_id by home_id
-    cons_prod_df["P_cons"] = 0
-    cons_prod_df["P_prod"] = 0
-    cons_prod_df["P_tot"] = 0
+    :param home_raw_data:   Dictionary of dataframes with the following format:
+                            {sensor_id1: df, sensor_id2: df, ...} where the df has
+                            columns -> ["sensor_id, day, ts, power"].
+    :param home_config:     The configuration for a specific home.
+
+    :return:                Return a dataframe with computed power data.
+    """
+    first_sid = home_config.index[0]
+    size = len(home_raw_data[first_sid])
+    cons_prod_df = pd.concat(
+        [
+            pd.Series([home_config.iloc[0, 0]] * size),
+            home_raw_data[first_sid]["day"],
+            home_raw_data[first_sid]["ts"],
+            pd.Series([0] * size),
```

Comment:
You don't need to duplicate by the size to assign a constant. You can also use `df.assign(column_name=value)`.

Reply:
The reason is that I don't have (yet) a dataframe, that's why I create a (huge) list with values otherwise there are nans in series

Reply:
`
grouped_sensors = home_raw_data.groupby('sensor_id')
first_group = grouped_sensors.get_group(home_raw_data.iloc[0, 0])
cons_prod_df = pd.DataFrame().assign(
        home_id=home_config.iloc[0, 0],
        day=first_group['day'],
        ts=first_group['ts'],
        P_cons=0,
        P_prod=0,
        P_tot=0
    )
for sid, sensor_df in grouped_sensors:
etc
`
I'm doing this but I have NaN for the home_id (others are good). I will investigate.
Is it necessary to apply a copy ?